### DISCUSSION TITLE
dynamic image in workflow

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#8676] [Q&A]
is it possible to have a workflow that has a dynamic middle step (with dynamic image)?

ie
if output of stepA is x then
step A>step B (running image x) >step C

but if output of stepA is y then
step A>step B (running image y)>step C

but if output of stepA is 'no-op' then just
step A>step C

the images x/y should not be hardcoded anywhere as i want to arbitrarily run any image name returned by the output of stepA

would also be good to have resources.limits for next step set based on output of earlier step

**Comment by terrytangyuan** [comment]
You can parameterize the resources to take the output from previous steps.

**Reply by tooptoop4** [reply]
@terrytangyuan I guess there are 3 parts to this: the decision to run step B at all, what image step B uses, resources of next step based on output of earlier step. do u have example covering all?

**Reply by terrytangyuan** [reply]
Use DAG/dependencies to decide whether to run a step. Use parameters to customize resources and image.
